python/pyarrow/array.pxi

```diff
@@ -21,28 +21,28 @@ import warnings
 
 cdef _sequence_to_array(object sequence, object mask, object size,
                         DataType type, CMemoryPool* pool, c_bool from_pandas):
-    cdef int64_t c_size
-    cdef PyConversionOptions options
+    cdef:
+        int64_t c_size
+        PyConversionOptions options
+        shared_ptr[CChunkedArray] chunked
 
     if type is not None:
         options.type = type.sp_type
 
     if size is not None:
         options.size = size
 
-    options.pool = pool
     options.from_pandas = from_pandas
-    options.ignore_timezone = os.environ.get('PYARROW_IGNORE_TIMEZONE', False)
```

Comment:
What's the reason this is being removed? It's handled elsewhere now?

Reply:
We still need it for 2.0.0, so reverting. Nice catch :)

Reply:
It wasn't catched by any tests?

Reply:
AFAIK It is only tested from the spark integration tests. I can add a unittest for it.

Reply:
Added a [unittest](https://github.com/apache/arrow/pull/8088/commits/29aa420f020233f29fd8ff225c58717e6c17b634#diff-7fddb4a3590947db6d84787eca4d80faR969-R998) for it, please check its validity.

Reply:
Thanks for the test! Looks good to me

Reply:
Thanks!